src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -453,120 +467,172 @@ impl<S: MetaStore> HummockManager<S> {
+                    .await?;
+                let mut config = self
+                    .compaction_group_manager
+                    .read()
+                    .await
+                    .get_default_compaction_group_config();
+                config.split_by_state_table = allow_split_by_table;
+
+                new_version_delta.group_deltas.insert(
+                    new_compaction_group_id,
+                    GroupDeltas {
+                        group_deltas: vec![GroupDelta {
+                            delta_type: Some(DeltaType::GroupConstruct(GroupConstruct {
+                                group_config: Some(config.clone()),
+                                group_id: new_compaction_group_id,
+                                parent_group_id,
+                                new_sst_start_id,
+                                table_ids: table_ids.to_vec(),
+                            })),
+                        }],
+                    },
+                );
 
+                new_group = Some(CompactionGroup {
+                    group_id: new_compaction_group_id,
+                    compaction_config: Arc::new(config),
+                });
+                new_version_delta.group_deltas.insert(
+                    parent_group_id,
+                    GroupDeltas {
+                        group_deltas: vec![GroupDelta {
+                            delta_type: Some(DeltaType::GroupMetaChange(GroupMetaChange {
+                                table_ids_remove: table_ids.to_vec(),
+                                ..Default::default()
+                            })),
+                        }],
+                    },
+                );
+                new_compaction_group_id
+            }
+        };
         let mut branched_ssts = BTreeMapTransaction::new(&mut versioning.branched_ssts);
         let mut trx = Transaction::default();
         new_version_delta.apply_to_txn(&mut trx)?;
         self.env.meta_store().txn(trx).await?;
+        if let Some(new_compaction_group) = new_group {
+            self.compaction_group_manager
+                .write()
+                .await
+                .compaction_groups
+                .insert(new_compaction_group.group_id, new_compaction_group);
+        }
         let sst_split_info = versioning
             .current_version
             .apply_version_delta(&new_version_delta);
         // Updates SST split info
-        for (object_id, sst_id, parent_old_sst_id, parent_new_sst_id) in sst_split_info {
+        for (object_id, sst_id, _parent_old_sst_id, parent_new_sst_id) in sst_split_info {
             match branched_ssts.get_mut(object_id) {
                 Some(mut entry) => {
-                    let p = entry.get_mut(&parent_group_id).unwrap();
-                    let parent_pos = p.iter().position(|id| *id == parent_old_sst_id).unwrap();
+                    assert!(parent_new_sst_id.is_some());
```

Comment:
seems we cannot assert here in our new design

Reply:
Yes. Because it may be a trivial-split file. I would fix this case.